tests/browser_reporting.js

```diff
@@ -0,0 +1,33 @@
+function reportResultToServer(result, sync, port) {
+  port = port || 8888;
+  if (reportResultToServer.reported) {
+    // Only report one result per test, even if the test misbehaves and tries to report more.
+    reportErrorToServer("excessive reported results, sending " + result + ", test will fail");
+  }
+  reportResultToServer.reported = true;
+
+  var xhr = new XMLHttpRequest();
+  if (typeof Module === 'object' && Module && Module['pageThrewException']) result = 12345;
```

Comment:
how about putting this expression in a var or function or something since it's duplicated below?